app/validators/ufn_validator.rb

```diff
@@ -0,0 +1,29 @@
+class UfnValidator < ActiveModel::EachValidator
+  attr_reader :record, :attribute
+
+  def validate_each(record, attribute, value)
+    # this will trigger a `:blank` error when `presence: true`
+    return if value.blank?
+
+    @record = record
+    @attribute = attribute
+
+    matched, *date_parts_strings = *value.match(%r{\A(\d{2})(\d{2})(\d{2})/\d{3}\z})
+    date_parts = date_parts_strings.reverse.map(&:to_i)
+
+    if matched && Date.valid_date?(*date_parts)
+      # convert to 4 digit date
+      date_parts[0] += 2000
```

Comment:
Is there _any_ possibility that the system will ever deal with some very old case from the 90s? I don't think so, but worth just checking. If so I guess we say any year that's... greater than the current year... plus arbitrary constant years... is from the 20th century...?